extensions/wikia/Recirculation/js/incontent.js

```diff
@@ -0,0 +1,187 @@
+/*global require*/
+require([
+	'jquery',
+	'wikia.window',
+	'wikia.abTest',
+	'wikia.nirvana',
+	'wikia.mustache',
+	'ext.wikia.recirculation.tracker',
+	'ext.wikia.recirculation.utils'
+], function ($, w, abTest, nirvana, Mustache, tracker, utils) {
+	// Currently only showing for English communities
+	if (w.wgContentLanguage !== 'en') { return; }
+
+
+	var $container = $('#mw-content-text'),
+		template;
+
+	function injectInContentWidget($container) {
+		var $links = $container.find('a'),
+			sections = buildSections($container.find('h2')),
+			width,
+			firstSuitableSection,
+			topTitles;
+
+		// If this page doesn't have enough content (either links or sections) we
+		// don't want to show this widget
+		if ($links.length < 8 || sections.length < 3) {
+			return;
+		}
+
+		// The idea is to show links above the first section under an infobox
+		width = $container.outerWidth();
+		firstSuitableSection = sections.find(function(item, index) {
+			return item.width === width;
+		});
+
+		if (!firstSuitableSection) {
+			return;
+		}
+
+		topTitles = findTopTitles($links);
+
+		if (topTitles.length < 3) {
+			return;
+		}
+
+		nirvana.sendRequest({
+			controller: 'ArticlesApi',
+			method: 'getDetails',
+			format: 'json',
+			type: 'get',
+			data: {
+				titles: topTitles.join(','),
+				abstract: 0,
+				width: 50,
+				height: 50
+			},
+			callback: renderArticles(firstSuitableSection)
+		});
+	}
+
+	function renderArticles(section) {
+		var placeholderImage = w.wgExtensionsPath + '/wikia/Recirculation/images/placeholder.png';
```

Comment:
That's a fair point and as always a trade-off: is it better to keep the path in JS to avoid logic in template or add not that much complicated logic to the template and keep path in SCSS ;) It's fine to think about it with next iteration since you want to release it only to small amount of users now.